docs/agda-spec/src/Spec/Protocol/Properties.agda

```diff
@@ -0,0 +1,101 @@
+{-# OPTIONS --safe #-}
+
+open import Spec.BaseTypes using (Nonces)
+open import Spec.BlockDefinitions
+open import Ledger.Crypto
+open import Ledger.Script
+open import Ledger.Types.Epoch
+open import Data.Rational.Ext
+
+module Spec.Protocol.Properties
+  (crypto : _) (open Crypto crypto)
+  (nonces : Nonces crypto) (open Nonces nonces)
+  (es     : _) (open EpochStructure es)
+  (ss     : ScriptStructure crypto es) (open ScriptStructure ss)
+  (bs     : BlockStructure crypto nonces es ss) (open BlockStructure bs)
+  (af     : _) (open AbstractFunctions af)
+  (rs     : _) (open RationalExtStructure rs)
+  where
+
+open import Data.Rational as ℚ using (1ℚ)
+open import Ledger.Prelude
+open import Spec.Protocol crypto nonces es ss bs af rs
+open import Spec.BaseTypes crypto using (OCertCounters)
+open import Spec.UpdateNonce crypto nonces es
+open import Spec.UpdateNonce.Properties crypto nonces es
+open import Spec.OperationalCertificate crypto nonces es ss bs af
+open import Spec.OperationalCertificate.Properties crypto nonces es ss bs af
+open import InterfaceLibrary.Common.BaseTypes crypto using (PoolDistr; lookupPoolDistr)
+
+private
+
+  checkLeaderVal? : ∀ cert int σ → Dec (checkLeaderVal cert int σ)
+  checkLeaderVal? (certℕ , certℕprf) (f , posf , f≤1) σ
+    with f ≟ 1ℚ
+  ... | yes _   = yes _
+  ... | no f≢1ℚ =
+        let
+          p = ℤ.pos certℕ ℚ./ (2 ^ 512)
```

Comment:
I'm not really sure, but at least that's what we had agreed upon when developing the LaTeX spec in #982. 😅 Do you think we should revise this?

Reply:
If the run time is fine, then it's fine as-is. I have no experience with exeuctable Agda etc, so I don't have any idea if the run time will be fine for this. We'll pay the price roughly twice per block, in a test, I think 🤷

Reply:
I'm not sure about the run time for this neither. Are you fine with leaving this as-is for now and addressing any potential issues later?

Reply:
For stupid reasons, the Agda stdlib needs to compute the gcd of the two values twice when doing division. And every other operation is defined in terms of division, so addition is even slower for example. But operations on natural numbers are relatively fast, so it might not be a problem.

Reply:
I just tested computing `show (1ℚ - ((+ 1000) / 2 ^ 512))` in the type checker. It's a fraction that doesn't fit on my screen in a single line and it was instant. So I wouldn't be too worried.

Reply:
That's very interesting. Thanks, Andre!